### PR TITLE
Fixing memory leak and clean up freeing operations

### PR DESCRIPTION
**Pull request by servos**
Main fix is to a memory leak that was present in the costmap_2d/inflation_layer. Also includes other clean up to explicitly free memory that was allocated and ensure memory is allocated when needed.

### CODE REVIEW

File: global_planner/src/dijkstra.cpp [hunk 1]
```diff
@@ -49,6 +49,15 @@ DijkstraExpansion::DijkstraExpansion(PotentialCalculator* p_calc, int nx, int ny
     priorityIncrement_ = 2 * neutral_cost_;
 }
 
+DijkstraExpansion::~DijkstraExpansion() {
+  printf("Dijkstra deconstruct \n");
```
Comment (DLu) [inline global_planner/src/dijkstra.cpp]: Do you have a strong reason for including this debug print?

File: costmap_2d/plugins/inflation_layer.cpp [hunk 2]
```diff
@@ -292,19 +311,25 @@ void InflationLayer::deleteKernels()
...
   if (cached_costs_ != NULL)
   {
     for (unsigned int i = 0; i <= cached_cell_inflation_radius_ + 1; ++i)
     {
-      delete[] cached_costs_[i];
+      if (cached_costs_[i])
+        delete[] cached_costs_[i];
     }
-    delete[] cached_costs_;
+    if (cached_costs_)
```
Comment (DLu) [inline costmap_2d/plugins/inflation_layer.cpp]: Is this necessary? We just checked if cached_costs_ was null.

File: costmap_2d/include/costmap_2d/inflation_layer.h [hunk 3]
```diff
@@ -194,4 +195,4 @@ class InflationLayer : public Layer
 
 }  // namespace costmap_2d
 
-#endif  // COSTMAP_2D_INFLATION_LAYER_H_
\ No newline at end of file
+#endif  // COSTMAP_2D_INFLATION_LAYER_H_
```
Comment (DLu) [inline costmap_2d/include/costmap_2d/inflation_layer.h:198]: In the future, please try not to include these nonfunctional changes in functional PRs.

**Comment by DLu** [comment]
Overall, looks good. Thanks for cleaning up the sloppiness.

**Comment by mikeferguson** [comment]
Looks like this has been updated, for reference @servos please post a comment in the future when you push new commits -- otherwise we don't actually get any notice that commits have been pushed.

I'll merge this later today -- I want to wait for the current CI job to finish before adding another commit.

**Pull request merged**